R/generics.R

```diff
@@ -81,7 +86,9 @@ setMethod("make_row_df", "MatrixPrintForm", function(tt, colwidths = NULL, visib
                                                      repr_inds = integer(),
                                                      sibpos = NA_integer_,
                                                      nsibs = NA_integer_,
-                                                     max_width = NULL) {
+                                                     max_width = NULL,
+                                                     fontspec = font_spec(),
+                                                     col_gap = mf_colgap(tt) %||% 3L) {
```

Comment:
I think we need a general rule for this. Deciding where it should be set and where it is only "transmitted" and if we keep it from object getters or not. At the moment I find it very hard to track the origin of a default value. I would suggest keeping a standard default (like NULL) until there are no more user-facing functions where if it is not set yet it is set from the object or a default

Reply:
I think it should be able to be set in any user-facing call, and should override the one carried around if that is the case. 

We could get rid of the `%||% 3L` part here if we want logic in the body that handles when `length(mf_colgap(tt)) == 0`, though honestly I kinda like this better since you can see what it does in the default value, whereas the othr way you have to look in the body of the function...

Reply:
I think it makes sense in this case. What do we add in (for example) export_txt ? We cannot reference the object tt because now it can be a list of tables/listings, so we should have either NULL and then we add what you added here in a `if(is.null)` inside `paginate_to_mpfs` or just always 3L/2L. I am leaning towards the second for this case, but then I am a bit confused whether we assign/retrieve from the object itself at all and how this depends on the creation of the object table/listing. We would need a graph of these dependencies. Many would respect the same pipeline I suspect

Reply:
So MPFs carry around their col_gaps after creation, and if it has one, I think that should be used, so I don't think a default of 3/2 taht overrides taht would be desirable